### DISCUSSION TITLE
File selector support?

### DISCUSSION THREAD

**Discussion by robert-winkler** [Igalia/cog#628] [Q&A] [answered]
Hi, cog is a great launcher for webmailing on mobian (on the PinePhone).

However, if I want to send an attachment, I touch the file selector button for uploading the file and nothing happens (downloading attachments works). I tries Posteo and Tutanota.

Any idea how to fix this?

**Comment by philn** [comment]
Cog is not meant to be a desktop web browser, it has very limited support for popups, currently only option menus are supported. If you need a desktop web browser I'd suggest WebKitGTK's Epiphany.

**Comment by robert-winkler** [comment]
Thanks for your comment! Anyway, cog is the best mobian app for webmailing 😉.

**Comment by aperezdc** [comment, marked as answer]
File selectors have been added to Cog's Wayland, X11 and GTK4 platform plug-ins in https://github.com/Igalia/cog/pull/606, the upcoming 0.20.x release series will have them when building with support for libportal.
